pages/_app.js

```diff
@@ -125,18 +147,48 @@ export default function App({ Component, pageProps }) {
       return;
     }
 
-    const newCategory = { name: capitalizeFirstLetter(category), id: ulid() };
-    setCategories([...categories, newCategory]);
-    setSuccessMessage("Category successfully added!");
+    const response = await fetch("/api/categories", {
+      method: "POST",
+      headers: {
+        "Content-Type": "applications/json",
+      },
+      body: JSON.stringify(correctFormat),
+    });
+
+    if (response.ok) {
+      mutateCategories();
+    }
   }
 
   function handleOpenEditModeCategory(category) {
     setOriginalCategoryName(category.name);
-    setIsEditCategory(category.id);
+    setIsEditCategory(category._id);
     setcategoryToEdit(category);
   }
 
-  function handleSaveEditCategory(editedCategory) {
+  async function handleSaveEditCategory(editedCategory) {
+    const response = await fetch(`/api/categories/${editedCategory._id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(editedCategory),
+    });
+
+    if (response.ok) {
+      mutateCategories(
+        categories.map((category) =>
+          category._id === editedCategory._id ? editedCategory : category
+        ),
+        false
+      );
```

Comment:
you only need to call `mutateCategories()` without passing a value. When you look at `mutateTransactions()` you already doing it corretly. calling this function simply refetches the data from the backend.